### PR TITLE
Rating curve model update to include uncertainty

### LINKED ISSUE #5: Parse qualitative discharge uncertainty
Discharge observations from NWIS are rated as excellent, good, fair, poor, or unspecified.
Assume 2-sigma intervals of
Excellent = 2%
Good = 5%
Fair = 8%
Poor = 12%
Unspecified = Poor

Start with the above, but to close this issue we should also consider the measurement type. The intervals above are for direct measurements. For all "indirect" measurements assume 20% error

### PR DESCRIPTION
**Pull request by kjdoore**
Updates the rating curve model to now allow for uncertainty on the discharge. Discharge uncertainty is also directly parsed from NWIS using the measurement quality code (resolves #5).

### CODE REVIEW

File: src/discontinuum/data_manager.py [hunk 1]
```diff
@@ -50,25 +51,31 @@ class DataManager:
     """ """
 
     target_pipeline: Type[Pipeline] = LogStandardPipeline
+    uncertainty_pipeline: Type[Pipeline] = None
```
Comment (thodson-usgs) [inline src/discontinuum/data_manager.py:54]: We can omit this in lieu of the error pipeline. Revise accordingly.

**Review by thodson-usgs — COMMENTED** [review]
Nice progress! Two comments.

File: src/rating_gp/models/gpytorch.py [hunk 2]
```diff
@@ -46,14 +46,17 @@ def __init__(
...
         # assume a constant measurement error for testing
-        noise = 0.1**2 * torch.ones(y.shape[0]).reshape(1, -1)
+        if y_unc is not None:
+            noise = y_unc
+        else:
+            noise = 0.1**2 * torch.ones(y.shape[0]).reshape(1, -1)
         self.likelihood = gpytorch.likelihoods.FixedNoiseGaussianLikelihood(
             noise=noise,
-            learn_additional_noise=False,
+            learn_additional_noise=True,
```
[the remark this answers is not in the document]
Comment (kjdoore) [reply src/rating_gp/models/gpytorch.py:59]: Okay, I tested this with it as `False` and was getting a warning when predicting, but we can always turn it back on if needed later.

File: src/rating_gp/pipeline.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+from numpy.typing import ArrayLike
+from sklearn.base import BaseEstimator, OneToOneFeatureMixin, TransformerMixin
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import FunctionTransformer, StandardScaler
+from xarray import DataArray
+from discontinuum.pipeline import MetadataManager
+
+
+
+class LogPropagation(OneToOneFeatureMixin, TransformerMixin, BaseEstimator):
```
Comment (thodson-usgs) [inline src/rating_gp/pipeline.py:14]: For our sanity, assume the `provider` supplies uncertainty as GSE. In other words, the provider will include a function to estimate GSE from % error. That way the current pipeline only needs to handle the log transformation. If we decide % error is correct, then we can pass % error from the provider directly to your LogPropogation pipeline.

**Comment by kjdoore** [comment]
@thodson-usgs
I made the updates to remove the `LogUncertaintyPipeline` for now. After thinking about it some more, if we assume the input uncertainty to the model is in GSE, then using the `LogErrorPipeline` is the correct method. However, if the uncertainty is SE, then the `LogUncertaintyPipeline` is correct. Finding the [clear definition of the GSE](https://en.wikipedia.org/wiki/Geometric_standard_deviation) helped me out there.

**Review by thodson-usgs — APPROVED** [review]
Thou shalt account for uncertainty in thy discharge - check!

**Pull request merged**